contrib/postgres-cmake/CMakeLists.txt

```diff
@@ -79,6 +79,17 @@ if (OS_DARWIN OR OS_FREEBSD OR USE_MUSL)
     target_compile_definitions(_libpq PRIVATE -DSTRERROR_R_INT=1)
 endif()
 
+# Mac 15.4 / Xcode 16.3 started defining srtchrnul
+# Could be removed after cleanup after fix in upstream https://www.postgresql.org/message-id/flat/385134.1743523038%40sss.pgh.pa.us
+if (OS_DARWIN AND NOT USE_MUSL)
+    try_compile(COMPILE_SUCCEEDED ${CMAKE_CURRENT_BINARY_DIR}/compile_tests ${CMAKE_CURRENT_SOURCE_DIR}/compile_tests/snprintf_check.c)
```

Comment:
`check_symbol_exists` is better, but it is forbidden - https://github.com/ClickHouse/ClickHouse/blob/979a47d3482d073d8ea6180e7f6da7fb44bca557/ci/jobs/scripts/check_style/check_cpp.sh#L316 and `try_compile` is the same, but very low level.

So I think that for now it is better to explicitly check for SDK version